client/thin-replica-client/CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+project("Thin Replica Client")
+
+# CMake helper modules
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/../cmake")
+
+find_package(OpenSSL REQUIRED)
+
+# Build the Thin Replica Client library
+include_directories(${CMAKE_CURRENT_BINARY_DIR})
+add_library(thin_replica_client SHARED
+  "src/thin_replica_client.cpp"
+  "src/trace_contexts.cpp"
+  "src/trs_connection.cpp"
+  "src/crypto_utils.cpp"
+  "src/trc_hash.cpp")
+target_include_directories(thin_replica_client PUBLIC include)
+target_compile_definitions(thin_replica_client PUBLIC USE_LOG4CPP)
+target_include_directories(thin_replica_client PUBLIC
```

Comment:
Does this need to be public? If someone depends on the secrets manager then they should include it directly.